### DISCUSSION TITLE
Incomplete completion in tsx-ts-mode by typescript language server

### DISCUSSION THREAD

**Discussion by skybreakerr** [emacs-lsp/lsp-mode#4255] [Q&A] [answered]
I can get completion everywhere, except in html attributes 
![image](https://github.com/emacs-lsp/lsp-mode/assets/76730713/acc91a41-1f15-4e18-a77f-bfef7154a856)
![image](https://github.com/emacs-lsp/lsp-mode/assets/76730713/fa4403e8-128b-4d6c-9983-2e9edcc47aaf)
config - http://0x0.st/H3Og.bin

**Comment by yyoncho** [comment]
If the server is not returning the completion items (e. g. if client side is not filtering them by mistake) then you should move the question in ts-ls server repo. Use M-x lsp-toggle-trace-io and then lsp-workspace-show-log and look for `textDocument/completion`

**Reply by skybreakerr** [reply, marked as answer]
Thanks for reply, found the issue was not having yasnippet
https://github.com/typescript-language-server/typescript-language-server/issues/679#issuecomment-1424550739
